pkg/storage/tsdb/config.go

```diff
@@ -441,6 +434,15 @@ func (cfg *BucketStoreConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.BucketIndex.RegisterFlagsWithPrefix(f, bucketIndexFlagPrefix)
 	cfg.IndexHeader.RegisterFlagsWithPrefix(f, "blocks-storage.bucket-store.index-header.")
 
+	// TODO: Remove in Mimir 2.12.
+	// Take value from Indexheader DeprecatedConfig if it is set
+	if cfg.DeprecatedIndexHeaderLazyLoadingEnabled {
+		cfg.IndexHeader.LazyLoadingEnabled = cfg.DeprecatedIndexHeaderLazyLoadingEnabled
+	}
```

Comment:
We're registring flags here, not parsing them, so no flags were set yet.

Reply:
Missed that fact. I moved to `Mimir.initStoreGateway`. Please advise if it is the good place for setting this or whether do we need to set this deprecated flag override in the first place.